test/test_synthetic.py

```diff
@@ -0,0 +1,407 @@
+import os, os.path as op
+from dataclasses import dataclass
+import yaml
+import subprocess
+from datetime import datetime
+import numpy as np
+import xarray as xr
+
+import RAiDER
+from RAiDER.llreader import BoundingBox
+from RAiDER.models.gmao import GMAO
+from RAiDER.models.weatherModel import make_weather_model_filename
+from RAiDER.losreader import Raytracing, getTopOfAtmosphere
+from RAiDER.utilFcns import lla2ecef, ecef2lla
+from RAiDER.cli.validators import modelName2Module
+
+
+import pytest
+from test import TEST_DIR
+
+
+def update_yaml(dct_cfg:dict, dst:str='temp.yaml'):
+    """ Write a new yaml file from a dictionary.
+
+    Updates parameters in the default 'raider.yaml' file.
+    Each key:value pair will in 'dct_cfg' will overwrite that in the default
+    """
+
+    template_file = os.path.join(
+                    os.path.dirname(RAiDER.__file__), 'cli', 'raider.yaml')
+
+    with open(template_file, 'r') as f:
+        try:
+            params = yaml.safe_load(f)
+        except yaml.YAMLError as exc:
+            print(exc)
+            raise ValueError(f'Something is wrong with the yaml file {template_file}')
+
+    params = {**params, **dct_cfg}
+
+    with open(dst, 'w') as fh:
+        yaml.safe_dump(params, fh,  default_flow_style=False)
+
+    return dst
+
+
+def update_model(wm_file:str, kind:str, wm_dir:str='weather_files_synth'):
+    """ Update the weater model file; keep kind to equation to test """
+    assert kind in 'hydro wet1 wet2', 'Set  kind to hydro, wet1, or wet2'
+    from RAiDER.models.gmao import GMAO
+    # initialize dummy wm to calculate constant delays
+    # any model will do as 1) all constants same 2) all equations same
+    model = op.basename(wm_file).split('_')[0].upper().replace("-", "")
+    Obj = modelName2Module(model)[1]()
+    ds = xr.open_dataset(wm_file)
+    t  = ds['t']
+    p  = ds['p']
+    e  = ds['e']
+    if kind == 'hydro':
+        p = t
+    elif kind == 'wet1':
+        e = t
+        Obj._k3 = 0
+    elif kind == 'wet2':
+        e = t**2
+        Obj._k2 = 0
```

Comment:
we should put these assumptions into the description of the function. 

i.e. when Hydo is kind we test the hydrostatic component of the equation P/T where P=T.

Reply:
Again, this would be helped if there was some references to model integrals and assumptions.